ecosystem/sep-0045.md

```diff
@@ -0,0 +1,431 @@
+| `address`       | `C...` string | The **Client Account** address (`C...`) that the **Client** wishes to authenticate with the **Server**.                                                                                                                                                                                                                                                                                                                    |
+| `home_domain`   | string        | A **Home Domain**. Servers that generate tokens for multiple **Home Domain**s can use this parameter to identify which home domain the **Client** hopes to authenticate with.                                                                                                                                                                                                                                              |
+| `client_domain` | string        | (optional) a **Client Domain**. Supplied by **Clients** that intend to verify their domain in addition to the **Client Account**. See [Verifying the Client Domain](#verifying-the-client-domain). **Servers** should ignore this parameter if the **Server** does not support **Client Domain** verification, or the **Server** does not support verification for the specific **Client Domain** included in the request. |
+
+Example:
+
+```
+GET https://auth.example.com/?address=CCIBUCGPOHWMMMFPFTDWBSVHQRT4DIBJ7AD6BZJYDITBK2LCVBYW7HUQ&home_domain=example.com
+```
+
+#### Response
+
+##### Success
+
+On success the endpoint must return `200 OK` HTTP status code and a JSON object with the following fields:
+
+- `authorization_entries`: a list of XDR-encoded `SorobanAuthorizationEntry`. It contains an entry for each of the
+  **Client Account**'s signers in addition to a signed entry from the **Home Domain Address** and optionally the
+  **Client Domain Address**. The entries are not guaranteed to be in any specific order.
+
+Each entry's `root_invocation` function is a `contract_fn` with the following with no additional sub invocations:
+
+- `contract_address` is the `WEB_AUTH_CONTRACT_ID` from the **Server**'s `stellar.toml`
+- `function_name` is `web_auth_verify`
+- `args`:
+  - `address` matches the `address` from the request
+  - `home_domain` matches the `home_domain` from the request
+  - `home_domain_address` matches the **Home Domain Address**
+  - `web_auth_domain` matches the **Server**'s domain
+  - `client_domain` matches the **Client**'s domain
+  - `client_domain_address` matches the **Client Domain Address**
```

Comment:
~Should there be a step in the 'authorization' flow that says the client must verify that the function arguments in each invocation match the arguments provided to the endpoint?~

Nevermind, I see we encourage the client to do this in the recommendation section. We should probably find a way to reference this section in the authentication flow section since it include important guidance.